core/src/main/scala/kafka/coordinator/transaction/TransactionStateManager.scala

```diff
@@ -60,6 +60,14 @@ class TransactionStateManager(brokerId: Int,
 
   this.logIdent = "[Transaction Log Manager " + brokerId + "]: "
 
+  type WriteTxnMarkers = (String, // transactionalId
+    Long, // pid
+    Short, // producerEpoch
+    TransactionState, // currentState
+    TransactionMetadata, // current metadata
+    Int) // coordinatorEpoch
+    => Unit
```

Comment:
Thanks @dguy! It's close to what I had in mind. :) I sketched what I had in mind here:

https://github.com/ijuma/kafka/commit/53450ead374ca2d94d563a57f37839b32afb8193

The main difference is that I named the class `TxnMarkers` instead of `WriteTxnMarkerArgs` and I dropped the type alias `WriteTxnMarkers` as it did not seem too useful any longer. The latter is definitely subjective, so if you think it helps, then seems fine to keep it.

I also noticed that the companion object for `TransactionStateManager` is still named `TransactionManager`, seems like that should be renamed too? Once the function becomes simpler, having a separate callback is redundant in many cases, so I removed those too. Finally, I removed some unused imports. Anyway, feel free to take a look and take any of the changes that make sense.